### PR TITLE
Multipatch post-processing in serial and parallel

### PR DESCRIPTION
**Pull request by anlavandier**
Give the classes `OutputManager` and `PostProcessManager` complete multipatch capabilities.

This is done by:
* Adding the possibility to save the domain decomposition in `OutputManager`.
* Almost completely revamping `PostProcessManager` with multipatch in mind.
* Adding multipatch tests in `psydac/api/tests/test_post_processing.py`.
* Changing the `Pushforward` class to allow for faster instantiation.

Other features:
* Bugfixes and removal of trailing whitespaces in `psydac/core/bsplines_pyccel.py` and `psydac/core/kernels.py`
* New and improved geometry files from branch `devel-mp-ddm` in `mesh/` folder
* Minor changes in `psydac/mapping/discrete.py`
* Minor changes in `psydac/utilities/vtk.py`
* SymPDE is restricted to be == 0.15.1

### CODE REVIEW

File: psydac/cad/tests/test_geometry.py [hunk 1]
```diff
@@ -147,6 +147,7 @@ def test_geometry_2d_4():
     geo.export('circle.h5')
 
 #==============================================================================
+@pytest.mark.xfail # Changes in SymPDE make this fail
```
Comment (yguclu) [inline psydac/cad/tests/test_geometry.py:150]: Do you know if this will be fixed in branch `devel-mp-ddm`?

Comment (anlavandier) [reply psydac/cad/tests/test_geometry.py:150]: The tests haven't been changed and they pass on Said's PR so I think that they will.

File: psydac/api/postprocessing.py [hunk 2]
```diff
@@ -139,19 +154,35 @@ class OutputManager:
...
+        List of the names of the saved spaces.
+
+    _comm : mpi4py.MPI.Intracomm or None
+
+    _mode : str in {'r', 'r+', 'w', 'w-', 'x', 'a'}
+
+    _fields_file : h5py.File
+
+    _spaces_types_to_str : dict
+        Dictionnary from Sympde space Datatypes
```
Comment (yguclu) [inline psydac/api/postprocessing.py:173]: Typo (7 x in this file): Dictionnary/ies -> Dictionary/ies

**Review by yguclu — CHANGES_REQUESTED** [review]
Great job @anlavandier!

If it is easy and convenient to do so, please avoid using lambda functions in file `pushforward.py`. Other than that, I only have minor comments.

File: psydac/core/kernels.py [hunk 3]
```diff
@@ -4320,17 +4320,17 @@ def pushforward_2d_l2(fields_to_push: 'float[:,:,:]', jac_dets: 'float[:,:]', pu
...
-        Values of the Jacobian determinant of the Mapping
+    met_dets: ndarray
+        Metric determinant of the mapping
 
     pushed_fields: ndarray
         Push forwarded fields
     """
     for i_f in range(pushed_fields.shape[2]):
-        pushed_fields[:, :, i_f] = fields_to_push[:, :, i_f] / jac_dets[:, :]
+        pushed_fields[:, :, i_f] = fields_to_push[:, :, i_f] / met_dets[:, :]
```
[the remark this answers is not in the document]
Comment (anlavandier) [reply psydac/core/kernels.py:4330]: The square root. I should probably change the name

File: psydac/api/postprocessing.py (hunk 2 above)
Comment (anlavandier) [reply psydac/api/postprocessing.py]: Fixed in [7802f17](https://github.com/pyccel/psydac/pull/230/commits/7802f1727d614941a1cd1caf1810a483d405921c)

File: psydac/core/kernels.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (anlavandier) [reply psydac/core/kernels.py]: Changed the names in this file and `psydac/feec/pushforward.py` to clearly differentiate between the Metric Determinant and its sqaure root. [7802f17](https://github.com/pyccel/psydac/pull/230/commits/7802f1727d614941a1cd1caf1810a483d405921c)

File: psydac/feec/pushforward.py [hunk 4]
```diff
@@ -12,110 +12,135 @@
...
+    global_ends : tuple of int
+        Index of the last cell of the domain
+        for each direction. This is simply
+        the tuple of ``ncell - 1`` in each direction.
+
+    grid_local : list of arrays
+        Grid that is local to the process.
+        This is necessary when the mapping is
+        an Analytical mapping that isn't the identity.
+        In serial this is the same as ``grid``.
```
[the remark this answers is not in the document]
Comment (anlavandier) [reply psydac/feec/pushforward.py]: Changed in [7802f17](https://github.com/pyccel/psydac/pull/230/commits/7802f1727d614941a1cd1caf1810a483d405921c)

File: psydac/feec/pushforward.py [hunk 5]
```diff
@@ -12,110 +12,135 @@
...
+    Parameters
     ----------
-    mapping : SplineMapping or Mapping
-        Mapping used to push-forward
-    
     grid : list of arrays
-        Grid on which fields and other quantities will be evaluated
-    
+        Grid on which fields and other quantities will be evaluated.
+        If it's a regular tensor grid, then it is expecetd to be
```
[the remark this answers is not in the document]
Comment (anlavandier) [reply psydac/feec/pushforward.py]: Fixed in [7802f17](https://github.com/pyccel/psydac/pull/230/commits/7802f1727d614941a1cd1caf1810a483d405921c)

File: psydac/core/kernels.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (anlavandier) [reply psydac/core/kernels.py]: And in [5e77365](https://github.com/pyccel/psydac/pull/230/commits/5e7736515a77a905aea830321c4bd60ffd7c8078)

**Review by yguclu — APPROVED** [review]

**Pull request merged**